### DISCUSSION TITLE
Convert/clone state (Proxy) into Object

### DISCUSSION THREAD

**Discussion by santiagopuentep** [pmndrs/valtio#553] [Q&A] [answered]
I'm trying to extract some state and do a copy of it to do some modifications independently from the proxy. If I use strucutredClone the browser throws an error.

How can we do a deep copy of a part of the state? It doesn't need to be a Proxy, I just need the object content of it.

**Comment by dai-shi** [comment]
```js
const state = proxy({ obj: ... })

snapshot(state.obj) // will create a plain object very efficiently, but it's frozen

import { cloneDeep } from 'lodash'
cloneDeep(state.obj) // will probably do without freezing
```

**Comment by santiagopuentep** [comment, marked as answer]
So using `snapshot()` can be used safely to "unproxy" objects? Without calling reactivity on the fields (like it happens when accessing `useSnapshot`'s object).

**Reply by dai-shi** [reply]
That's correct. `snapshot()` also takes care of `ref()`, while `cloneDeep` doesn't.

**Reply by santiagopuentep** [reply]
Awesome. Thanks for answering so fast.
